framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -1223,6 +1228,11 @@ public int expunge(final SearchCriteria<T> sc) {
             str.append(sc.getWhereClause());
         }
 
+        if (limit > 0) {
```

Comment:
Use `addFilter` method with filter to add limit. This will also allow to add order by in the Delete queries.

Reply:
Could you give an example where we would want to order the deletes? This would only make the deletes even slower.

Reply:
@JoaoJandre If you add order by and limit on a query, it can be faster if there is an index on the order by column. This depends on the query plan and the data distribution.

Also, while deleting stats, it would be better to remove the older entries first. Otherwise the user might see some missing timestamps if they fetch the data while it's being deleted.

Reply:
I see how the index might be beneficial to SELECTs; however, we're only talking about DELETEs here, do you have any test cases using DELETEs?